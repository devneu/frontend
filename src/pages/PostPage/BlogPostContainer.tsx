import { connect } from 'react-redux';
import BlogPost, { OwnProps } from './BlogPost';
import { InitialState } from '../../reducers';
import BlogSelectors from '../../selectors/blog';
import { Post } from '../../types/blogTypes';
import { BlogActions } from '../../actions/BlogActions';

export interface StateProps {
  post?: Post | null;
}
export interface DispatchProps {
  addComment: (value: string, parentId: number | null) => void;
}

function selectPostById(state: InitialState, id: number) {
  const posts = BlogSelectors.postsSelector(state);
  return posts.find((post: Post) => post.postId === id);
}

const postId = 1;
function mapStateToProps(state: InitialState) {
  return {
    post: selectPostById(state, postId),
  };
}
const mapDispatchToProps = {
  addComment: (value: string, parentId: number | null) => BlogActions.addComment({
    value,
    postId,
    parentId,
  }),
};

export default connect<StateProps, DispatchProps, OwnProps, InitialState>(
  mapStateToProps,
  mapDispatchToProps,
)(BlogPost);
