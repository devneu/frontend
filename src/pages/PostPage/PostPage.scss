.container {
  max-width: 1500px;
  margin: 20px auto;

  .sidebar-blog-post {
    margin-left: 20px;
  }
}