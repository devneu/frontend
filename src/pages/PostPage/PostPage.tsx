import React from 'react';
import './PostPage.scss';
import { Col, Layout, Row } from 'antd';
import Post from '../../components/Post/Post';
import SearchWidget from '../../components/Search/Search';
import CategoriesWidget from '../../components/Categories/Categories';

const PostPage = () => (
  <Layout className="post-page container">
    <Row justify="center">
      <Col span={14}>
        <Post />
      </Col>
      <Col span={6} className="sidebar">
        <SearchWidget />
        <CategoriesWidget />
      </Col>
    </Row>
  </Layout>
);

export default PostPage;
