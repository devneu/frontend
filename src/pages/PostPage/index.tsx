import BlogPostContainer from './PostPageContainer';

export default BlogPostContainer;
