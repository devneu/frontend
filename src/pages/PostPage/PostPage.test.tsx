import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ReactRouter from 'react-router';
import PostPage from './PostPage';

jest.mock('react-markdown', () => () => 'MockReactMarkDown');

describe('PostPage', () => {
  it('Should successfully render', () => {
    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({});
    jest.spyOn(ReactRouter, 'useHistory').mockReturnValue({ push: jest.fn() } as any);
    const wrapper = renderer
      .create(
        <BrowserRouter>
          <PostPage />
        </BrowserRouter>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
