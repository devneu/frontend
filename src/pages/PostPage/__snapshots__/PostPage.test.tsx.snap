// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPage Should successfully render 1`] = `
<div
  className="post-page container"
>
  <section
    className="ant-layout ant-layout-has-sider layout"
  >
    <main
      className="ant-layout-content content"
    />
    <aside
      className="ant-layout-sider ant-layout-sider-light sidebar"
      style={
        Object {
          "flex": "0 0 400px",
          "maxWidth": "400px",
          "minWidth": "400px",
          "width": "400px",
        }
      }
    >
      <div
        className="ant-layout-sider-children"
      >
        <div
          className="ant-card ant-card-bordered search-widget-card"
        >
          <div
            className="ant-card-head"
            style={Object {}}
          >
            <div
              className="ant-card-head-wrapper"
            >
              <div
                className="ant-card-head-title"
              >
                Search
              </div>
            </div>
          </div>
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <form
              className="ant-form ant-form-inline"
              onReset={[Function]}
              onSubmit={[Function]}
            >
              <div
                className="ant-row ant-form-item"
                style={Object {}}
              >
                <div
                  className="ant-col ant-form-item-control"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <input
                        className="ant-input input-search"
                        id="searchInput"
                        name="search"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="Enter search term..."
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="ant-row ant-form-item"
                style={Object {}}
              >
                <div
                  className="ant-col ant-form-item-control"
                  style={Object {}}
                >
                  <div
                    className="ant-form-item-control-input"
                  >
                    <div
                      className="ant-form-item-control-input-content"
                    >
                      <button
                        className="ant-btn ant-btn-primary button-search"
                        onClick={[Function]}
                        type="submit"
                      >
                        <span>
                          Go!
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          className="ant-card ant-card-bordered categories-widget-card"
        >
          <div
            className="ant-card-head"
            style={Object {}}
          >
            <div
              className="ant-card-head-wrapper"
            >
              <div
                className="ant-card-head-title"
              >
                Categories
              </div>
            </div>
          </div>
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <div
              className="ant-row ant-row-start"
              style={
                Object {
                  "marginBottom": -4,
                  "marginLeft": -4,
                  "marginRight": -4,
                  "marginTop": -4,
                }
              }
            >
              <div
                className="ant-col ant-col-12"
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <a
                  className="ant-btn ant-btn-link"
                  href="#"
                  onClick={[Function]}
                >
                  <span>
                    Web Design
                  </span>
                </a>
              </div>
              <div
                className="ant-col ant-col-12"
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <a
                  className="ant-btn ant-btn-link"
                  href="#"
                  onClick={[Function]}
                >
                  <span>
                    HTML
                  </span>
                </a>
              </div>
              <div
                className="ant-col ant-col-12"
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <a
                  className="ant-btn ant-btn-link"
                  href="#"
                  onClick={[Function]}
                >
                  <span>
                    Freebies
                  </span>
                </a>
              </div>
              <div
                className="ant-col ant-col-12"
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <a
                  className="ant-btn ant-btn-link"
                  href="#"
                  onClick={[Function]}
                >
                  <span>
                    JavaScript
                  </span>
                </a>
              </div>
              <div
                className="ant-col ant-col-12"
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <a
                  className="ant-btn ant-btn-link"
                  href="#"
                  onClick={[Function]}
                >
                  <span>
                    CSS
                  </span>
                </a>
              </div>
              <div
                className="ant-col ant-col-12"
                style={
                  Object {
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
              >
                <a
                  className="ant-btn ant-btn-link"
                  href="#"
                  onClick={[Function]}
                >
                  <span>
                    Tutorials
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-card ant-card-bordered side-widget"
        >
          <div
            className="ant-card-head"
            style={Object {}}
          >
            <div
              className="ant-card-head-wrapper"
            >
              <div
                className="ant-card-head-title"
              >
                Side Widget
              </div>
            </div>
          </div>
          <div
            className="ant-card-body"
            style={Object {}}
          >
            You can put anything you want inside of these side widgets. They are easy to use, and feature the Bootstrap 5 card component!
          </div>
        </div>
      </div>
    </aside>
  </section>
</div>
`;
